### PR TITLE
v9: Set log level to `warning` by default 

### LINKED ISSUE #2816: [v9]: Set default log level to `SentryLevel.warning` if debug is enabled
### Description

Part of https://github.com/getsentry/sentry-dart/issues/2752 to only show meaningful logs

Related: https://github.com/getsentry/sentry-cocoa/issues/4467

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description

Update logging min level

## :bulb: Motivation and Context

Closes https://github.com/getsentry/sentry-dart/issues/2816

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_options.dart [hunk 1]
```diff
@@ -155,11 +159,13 @@ class SentryOptions {
 
   set debug(bool newValue) {
     _debug = newValue;
-    if (_debug == true && logger == noOpLogger) {
-      _logger = _debugLogger;
+    if (_debug == true &&
+        (logger == noOpLogger || diagnosticLogger?.logger == noOpLogger)) {
```
Comment (denrase) [inline dart/lib/src/sentry_options.dart:163]: Why the additional `diagnosticLogger?.logger == noOpLogger)` here? Removed and running `sentry_options_test.dart` all tests pass, so we are not testing this condition or it is always the same to logger.

**Review by denrase — CHANGES_REQUESTED** [review]

File: dart/lib/src/sentry_options.dart (hunk 1 above)
Comment (buenaflor) [reply dart/lib/src/sentry_options.dart:163]: if you change the debug flag multiple times the logger will only be wrapped by the diagnosticLogger and will neither be noOpLogger or debugLogger directly. so this is basically a way to handle the edge case where the user sets the debug to true and false multiple times. it's an edge case though

Comment (denrase) [reply dart/lib/src/sentry_options.dart:163]: Ah i see, could we add a test for this as well if it's not already there?

**Review by denrase — APPROVED** [review]
